packages/extension-seo-url-mapping/src/setup-app.ts

```diff
@@ -0,0 +1,168 @@
+// after this one. Only if no module is found will the referenced module in this project be used.
+import sampleHOC from 'overridable!./components/sample-hoc'
+
+// Others
+import extensionMeta from '../extension-meta.json'
+
+interface StoreSlice {
+    count: number
+    increment: () => void
+    decrement: () => void
+}
+
+// This is safe to delete if your extension does not use state. If you aren't using this, ensure you remove the
+// `withApplicationExtensionStore` usage below as well.
+const sliceInitializer: SliceInitializer<StoreSlice> = (set) => ({
+    count: 0,
+    increment: () => set((state) => ({count: state.count + 1})),
+    decrement: () => set((state) => ({count: state.count - 1}))
+})
+
+const isServerSide = typeof window === 'undefined'
+
+class Sample extends ApplicationExtension<Config> {
+    static readonly id = extensionMeta.id
+
+    /**
+     * Use this method to wrap or enhance your PWA-Kit application using [React higher-order components](https://legacy.reactjs.org/docs/higher-order-components.html).
+     * You can use this to add visual treatments to your application, change the props that are supplied to the application component
+     * or add things like providers and contexts to be used throughout your app.
+     */
+    extendApp<T extends React.ComponentType<T>>(
+        App: React.ComponentType<T>
+    ): React.ComponentType<T> {
+        const HOCs = [
+            // Example higher-order component, this can be safely removed.
+            sampleHOC,
+            // Optionally include state for this extension using `withApplicationExtensionStore`
+            (component: React.ComponentType<any>) =>
+                withApplicationExtensionStore(component, {
+                    id: extensionMeta.id,
+                    initializer: sliceInitializer
+                })
+        ]
+
+        return applyHOCs(App, HOCs)
+    }
+
+    /**
+     * This method is used to make changes to the PWA-Kit application routes. If your extension adds a new page to the application
+     * then you can add it to the router here. The routes passed to this method is an accrued list of routes that have been added
+     * from extensions applied before it. It is called during the `getRoutes` phase on both the server and the client.
+     *
+     * NOTE: If you instead want to modify a list of all the routes, refer to the `beforeRouteMatch` below.
+     */
+    extendRoutes(routes: RouteProps[]): RouteProps[] {
+        return routes
+    }
+
+    /**
+     * This method is used on the server during the rendering pipeline. It's provided a list of all the routes that your application
+     * is configured with, including those defined in the base application and those added by all the extensions. You can use this
+     * method to modify these routes in any way you want, but you must return an array of routes as a result.
+     */
+    beforeRouteMatch(allRoutes: RouteProps[]): RouteProps[] {
```

Comment:
@bendvc here is the bulk of the changes where I update the `beforeRouteMatch` to get the routes from `window.__CONFIG__.app.routes` if it's client side and `allRoutes` if it's server